sql, ui: include distsql queries in `sql.{service,exec}.latency` metrics

vilterp opened the pull request:
Fixes #31093

Previously, the `sql.{service,exec}.latency` histogram metrics only
included local queries; distributed queries' latencies were recorded in
the `sql.distsql.{service,exec}.latency` metrics.

However, the UI had assumed that the `sql.service.latency` included all
SQL statements both local and distributed, and therefore had a graph of
it on the Overview dashboard, as well as having its most recent value
in the sidebar of the time series area.

With this change, the latencies of all statements are recorded in
`sql.{service,exec}.latency`, changing those metrics to mean what we had
assumed they meant. DistSQL queries will no longer be left out of this
prominent graph and sidebar metric.

Release note (admin ui change): The "service latency: {90,99}th
percentile" graphs on the Overview and SQL dashboard, as well as the P50
and P99 latency numbers in the time series area sidebar, now reflect
latencies of both local and distributed queries. Previously, they only
included local queries.

The second commit could also just remove those graphs entirely.

Commit:
sql: include distsql queries in `sql.{service,exec}.latency` metrics

Previously, the `sql.{service,exec}.latency` histogram metrics only
included local queries; distributed queries' latencies were recorded in
the `sql.distsql.{service,exec}.latency` metrics.

However, the UI had assumed that the `sql.service.latency` included all
SQL statements both local and distributed, and therefore had a graph of
it on the Overview dashboard, as well as having its most recent value
in the sidebar of the time series area.

With this change, the latencies of all statements are recorded in
`sql.{service,exec}.latency`, changing those metrics to mean what we had
assumed they meant. DistSQL queries will no longer be left out of this
prominent graph and sidebar metric.

Release note (admin ui change): The "service latency: {90,99}th
percentile" graphs on the Overview and SQL dashboard, as well as the P50
and P99 latency numbers in the time series area sidebar, now reflect
latencies of both local and distributed queries. Previously, they only
included local queries.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31098)
<!-- Reviewable:end -->

jordanlewis reviewed (approved):
LGTM, quick gut check from @andreimatei would be good too - maybe there's some reason why this wasn't done at first?

andreimatei commented:
LGTM.
The changed code probably comes from a time where the engines were thoroughly distinct in the public psyche.

couchand commented on pkg/ui/src/views/cluster/containers/nodeGraphs/dashboards/sql.tsx:
```diff
@@ -138,7 +138,7 @@ export default function (props: GraphDashboardProps) {
 
     <LineGraph
-      title="Service Latency: DistSQL, 99th percentile"
+      title="Service Latency: Distributed SQL, 99th percentile"
```
I'd be in favor of just dropping these charts. @piyush-singh @Amruta-Ranade thoughts?

vilterp replied:
I agree. Given that you have the statements page if you want more of a breakdown, I don't think users really need this. The metric is still being recorded; it's accessible via the custom graph page.

piyush-singh replied:
Agreed, especially since we have made a concerted effort to remove language that mentions DistSQL elsewhere. Good to remove on my end!

vilterp replied:
Cool, thanks. They're gone!

Commit:
ui: remove DistSQL latency graphs from SQL dashboard

These are not very useful now that the execution engines have been
merged.

Release note: none

vilterp commented:
bors r+

Doubt that the `TestStoreRangeMergeWatcher` failure is related to this.

vilterp commented:
bors r+

vilterp commented:
bors r+
